### ISSUE TITLE
Enhanced UPE - Blocks integration

### ISSUE THREAD

**Issue by FangedParakeet** [Automattic/woocommerce-payments#5020]
### Description

Stripe will soon release a new refactor of their UPE offering new payment integration flows. The most crucial addition is that they will now offer server side payment intent confirmation for the UPE, meaning that the payment intent is no longer required prior to presenting the UI element and the intent can be created and confirmed after collecting the payment information into a payment method.

This necessitates a refactor for both shortcode and blocks checkouts, where both create a payment intent as the first order of business, before loading any UI elements. This new refactored blocks implementation may now have significant resemblance to the previous legacy card element implementation. This issue is to complete this blocks integration. 

This issue is currently blocked until Stripe accepts us into their beta release program for their new enhanced UPE tool. This is also likely blocked by #5019. Although it would not be impossible for these two pieces to be completed in parallel, there will likely be significant overlap on the back-end portions of both implementations, so it would be preferable to complete the PoC prior to this effort. 

### Acceptance criteria

* Able to successfully complete checkout with new payment method via Blocks checkout.

### Additional context

[Stripe UPE – Options to solve WooPay compatibility and WC Pay loss of control over the payments UI/UX](pc2DNy-2e2-p2)
#4603

**Comment by FangedParakeet** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/5020?planning-poker) with Zenhub @mdmoore @timur27

**Comment by timur27** [comment]
I think we need to update the description and the ACs.

From a technical standpoint, the success of this issue would be the working Blocks checkout page. However, we need to consider which payment methods should be working after this issue is completed. In addition to card and non-card gateways, there is also WooPay and Stripe Link. Should we include them both in the scope? They also need to be implemented for the shortcode, so a separate issue to cover Stripe Link & WooPay for both shortcode and Blocks seems to be a good option to consider as well.

Furthermore, any reusable components from issue #5869 should be re-used in this issue, and we can emphasize this in the ACs as well.
